grafana-plugin/src/pages/integrations/Integrations.tsx

```diff
@@ -486,19 +497,23 @@ class Integrations extends React.Component<IntegrationsProps, IntegrationsState>
     this.setState({ confirmationModal: undefined });
   };
 
-  handleIntegrationsFiltersChange = (integrationsFilters: Filters) => {
-    this.setState({ integrationsFilters }, () => this.debouncedUpdateIntegrations());
+  handleIntegrationsFiltersChange = (integrationsFilters: Filters, isOnMount: boolean) => {
+    this.setState({ integrationsFilters }, () => this.debouncedUpdateIntegrations(isOnMount));
   };
 
-  applyFilters = () => {
+  applyFilters = async (isOnMount: boolean) => {
     const { store } = this.props;
     const { alertReceiveChannelStore } = store;
     const { integrationsFilters } = this.state;
 
-    return alertReceiveChannelStore.updatePaginatedItems(integrationsFilters).then(() => {
-      this.setState({ page: 1 });
-      LocationHelper.update({ p: 1 }, 'partial');
-    });
+    const newPage = isOnMount ? store.getCurrentPage(PAGE.Integrations) : 1;
+
+    return alertReceiveChannelStore
+      .updatePaginatedItems(integrationsFilters, newPage, false, () => this.invalidateRequestFn(newPage))
+      .then(() => {
+        store.setCurrentPage(PAGE.Integrations, newPage);
```

Comment:
did not get why do we need `store.setCurrentPage` to navigate we use ` history.push(`${PLUGIN_ROOT}/foo` or react router links, to get current page we can use `getMatchedPage`

Reply:
`store.setCurrentPage` is not for navigation, is for updating the store to know which was the last page we hit (the page num, not the page string)